sensorlib/src/sensor/SemanticLidarSensor.py

```diff
@@ -519,3 +522,28 @@ def update_object_frame_and_timestamps_from_hitpoint(self, obj, timestamp):
                        timestamp=timestamp,
                        position=new_position
                        )
+
+    def update_velocity(self, detected_objects):
+        for detected_object in detected_objects:
+            if detected_object.objectId in prev_objects:
+                time_diff = detected_object.timestamp - prev_objects[detected_object.objectId]['timestamp']
+
+                if time_diff and np.all(detected_object.velocity==0):
+                    pose_diff_x = ((detected_object.position[0] - prev_objects[detected_object.objectId]['pose_x'])**2)**0.5
+                    pose_diff_y = ((detected_object.position[1] - prev_objects[detected_object.objectId]['pose_y'])**2)**0.5
+                    pose_diff_z = ((detected_object.position[2] - prev_objects[detected_object.objectId]['pose_z'])**2)**0.5
+
+                    detected_object.velocity[0] = pose_diff_x / time_diff
+                    detected_object.velocity[1] = pose_diff_y / time_diff
+                    detected_object.velocity[2] = pose_diff_z / time_diff
```

Comment:
Let's also calculate the angular velocity based on its orientation with a note in the comment saying "Calculation based on an assumption that the object's orientation is the direction of travel"